neon_data_models/models/api/llm.py

```diff
@@ -0,0 +1,194 @@
+# 3. Neither the name of the copyright holder nor the names of its
+#    contributors may be used to endorse or promote products derived from this
+#    software without specific prior written permission.
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+# CONTRIBUTORS  BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
+# OR PROFITS;  OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+# LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+# NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE,  EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from typing import List, Tuple, Optional, Literal
+from pydantic import Field, model_validator, computed_field
+
+from neon_data_models.models.base import BaseModel
+
+
+_DEFAULT_MQ_TO_ROLE = {"user": "user", "llm": "assistant"}
+
+
+class LLMPersona(BaseModel):
+    name: str = Field(description="Unique name for this persona")
+    description: Optional[str] = Field(
+        None, description="Human-readable description of this persona")
+    system_prompt: str = Field(
+        None, description="System prompt associated with this persona. "
+                          "If None, `description` will be used.")
+    enabled: bool = Field(
+        True, description="Flag used to mark a defined persona as "
+                          "available for use.")
+    user_id: Optional[str] = Field(
+        None, description="`user_id` of the user who created this persona.")
+
+    @model_validator(mode='after')
+    def validate_request(self):
+        assert any((self.description, self.system_prompt))
+        if self.system_prompt is None:
+            self.system_prompt = self.description
+        return self
+
+    @computed_field
+    @property
+    def id(self) -> str:
+        persona_id = self.name
+        if self.user_id:
+            persona_id += f"_{self.user_id}"
+        return persona_id
+
+
+class LLMRequest(BaseModel):
+    query: str = Field(description="Incoming user prompt")
+    # TODO: History may support more options in the future
+    history: List[Tuple[Literal["user", "llm"], str]] = Field(
+        description="Formatted chat history (excluding system prompt). Note "
+                    "that the roles used here will differ from those used in "
+                    "OpenAI-compatible requests.")
+    persona: LLMPersona = Field(
+        description="Requested persona to respond to this message")
+    model: str = Field(description="Model to request")
+    max_tokens: int = Field(
+        default=512, ge=64, le=2048,
```

Comment:
Latest LLMs in some cases can stabily produce response up to 8K tokens
But 2K is very reliable upper bound